### PR TITLE
Support rocksdb cloud and aws s3

### PR DESCRIPTION
**Pull request by panda-sheep**
As title

### CODE REVIEW

File: project/externals/aws-sdk-cpp.cmake [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        TMP_DIR ${BUILD_INFO_DIR}
+        STAMP_DIR ${BUILD_INFO_DIR}
+        DOWNLOAD_DIR ${DOWNLOAD_DIR}
+        #PATCH_COMMAND patch -p1 < ${CMAKE_SOURCE_DIR}/patches/${name}-2022-9-19.patch
+        SOURCE_DIR ${source_dir}
+        CMAKE_ARGS
+            ${common_cmake_args}
+            -DCMAKE_BUILD_TYPE=Release
+            -DBUILD_TESTS=OFF
+            -DENABLE_TESTING=OFF
```
Comment (wenhaocs) [inline project/externals/aws-sdk-cpp.cmake:26]: aws-sdk is too large. We don't need all of them. Maybe you can limit the libraries to the ones we need at this moment with:
'-DBUILD_ONLY=s3;kinesis;transfer'

Comment (panda-sheep) [reply project/externals/aws-sdk-cpp.cmake:24]: There is a problem with compiled packages when using the DBUILD_ONLY option.
Do not use this option for now

**Comment by panda-sheep** [comment]
@wenhaocs  test passed

**Review by wenhaocs — APPROVED** [review]
Have created an issue to build only the necessary components: #1372. Good for now!

**Pull request merged**